core/src/main/java/fr/sncf/osrd/new_infra/implementation/reservation/ReservationInfraBuilder.java

```diff
@@ -102,6 +106,23 @@ private ReservationInfra build() {
         return networkBuilder.build();
     }
 
+    /** Returns true if the route is controlled (requires explicit reservation to be used) */
+    private boolean isRouteControlled(ImmutableList<TrackRangeView> trackRanges) {
+        // TODO: eventually, add an optional parameter to RJSRoute that would override this behavior
+        for (int i = 1; i < trackRanges.size(); i++) {
+            var prev = trackRanges.get(i - 1).track;
+            var next = trackRanges.get(i).track;
+            if (prev != next) {
+                var neighbors = prev.getEdge().getNeighbors(EdgeEndpoint.endEndpoint(prev.getDirection()));
```

Comment:
I'm confused as to why this is needed, isn't just checking if the edge is a switchbranch enough?